udp/src/main/scala/akka/stream/alpakka/udp/impl/UdpBind.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.udp.impl
+
+import java.net.InetSocketAddress
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.io.{IO, Udp}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.stream.alpakka.udp.UdpMessage
+import akka.stream.stage._
+
+import scala.concurrent.{Future, Promise}
+
+class UdpBindLogic(localAddress: InetSocketAddress, boundPromise: Promise[InetSocketAddress])(
```

Comment:
A top level comment for type would be great. 

Should be private/marked internal?